src/scripts/main.js

```diff
@@ -1 +1,74 @@
 'use strict';
+
+const promise1 = new Promise((resolve, reject) => {
+  const timeout = setTimeout(reject, 3000);
+
+  document.addEventListener('click', () => {
+    clearTimeout(timeout);
+    resolve();
+  });
+});
+const promise2 = new Promise((resolve) => {
+  document.addEventListener('contextmenu', (even) => {
+    even.preventDefault();
+    resolve();
+  });
+
+  document.addEventListener('click', resolve);
```

Comment:
Logical issue: The `click` event listener will resolve the promise even if it was already resolved by the `contextmenu` event. Consider adding a check to ensure the promise is resolved only once.